lib/syslogger/logger.rb

```diff
@@ -4,11 +4,11 @@ module SysLogger
   class Logger < MonoLogger
     attr_reader :logdev, :default_formatter
 
-    def initialize(logdev = nil, shift_age = 0, shift_size = 1048576, &block)
-      if logdev.nil? && block_given?
-        super(SysLogger::IO.new(&block), shift_age, shift_size)
+    def initialize(logdev = nil, shift_age = 0, shift_size = 1048576)
+      if logdev.is_a?(Proc)
+        super(SysLogger::IO.new(logdev), shift_age, shift_size)
```

Comment:
Does this actually work? I think it needs to be `Syslogger::IO.new(&logdev)`. Anyways, I would rather just keep the syntax we have. Seems more standard to just accept a block that creates the `logdev`.